### DISCUSSION TITLE
How do I update a value in place while using the update function?

### DISCUSSION THREAD

**Discussion by mohnish7** [supabase/supabase#7965] [Questions]
Suppose table `A` has a field `B`, which stores the value `1` and id `abc123`. Typically, in order to update the value in `B` to say, `2`, I would do 

`supabase_client.table("A").update({"B": 2}).eq("id", "abc123").execute()`

How would I go on about programmatically changing the value in field `B`? For example, I would like to increase the value by 1, and do something of the following sorts:

`supabase_client.table("A").update({"B": current_value + 1}).eq("id", "abc123").execute()`

Is there a way to do this in a very simple way? Thanks!

**Comment by GaryAustin1** [comment]
Right now you need to use an rpc call to a function to increment.   There is work underway to allow increment directly in PostgREST I believe, but have not seen a timeline.

**Comment by wcooperscott** [comment]
This can be done quite easily.

Create a SQL function that increments. You can read how to do that from [this answer](https://stackoverflow.com/a/10233360) on stack overflow. You may want the function to have an input parameter "id", so you can identify the correct row in update function.

And as to how to quite literally create a SQL function on Supabaase, you read the documentation for that [here](https://supabase.com/docs/guides/database/functions).

You can read on the documentation that explains how to execute a function [here](https://supabase.com/docs/reference/javascript/rpc). Here would be a code example of your client:
this.supabase.rpc("my_increment_function", { id: "abc123" })

And you can refer to the stack overflow answer for an example of SQL code.

Wrote this answer on my iPhone. Please excuse my brevity.
